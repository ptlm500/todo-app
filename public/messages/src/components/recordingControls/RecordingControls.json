[
  {
    "id": "todo.recControl.stop",
    "description": "Stop recording button text",
    "defaultMessage": "Stop"
  },
  {
    "id": "todo.recControl.start",
    "description": "Start recording button text",
    "defaultMessage": "Start"
  },
  {
    "id": "todo.recControl.clear",
    "description": "Clear recording button text",
    "defaultMessage": "Clear"
  }
]