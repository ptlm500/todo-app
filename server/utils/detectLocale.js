const acceptLanguage = require('accept-language');
const config = require('../config.json');

function detectLocale(req) {
  acceptLanguage.languages(config.ACCEPTED_LANGUAGES);

  const cookieLocale = req.cookies.locale;

  return acceptLanguage.get(cookieLocale || req.headers['accept-language']) ||
    config.DEFAULT_LOCALE;
}

module.exports = detectLocale;
