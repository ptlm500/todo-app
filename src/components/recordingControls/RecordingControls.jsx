import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import styled from 'styled-components';
import {
  toggleRecording,
  clearRecording,
  back,
  forward
} from '../../state/actions/actions';
import { SecondaryButton } from '../button';
import Indicator from '../indicator/Indicator';
import { theme } from '../../styles';

function RecordingControls(props) {
  const {
    className,
    recordingEnabled,
    disableForward,
    disableClear,
    onToggleRecording,
    onClearRecording,
    onBack,
    onForward
  } = props;

  return (
    <div className={className}>
      <Indicator pulsing fill={theme.red} disabled={!recordingEnabled} />
      <SecondaryButton onClick={onToggleRecording}>
        {recordingEnabled ? 'Stop' : 'Start'}
      </SecondaryButton>
      <SecondaryButton danger onClick={onClearRecording} disabled={disableClear}>
        Clear
      </SecondaryButton>
      <SecondaryButton onClick={onBack}>
        Back
      </SecondaryButton>
      <SecondaryButton onClick={onForward} disabled={disableForward}>
        Forward
      </SecondaryButton>
    </div>
  );
}

export const StyledRecordingControls = styled(RecordingControls)`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  left: 0;
  bottom: 0;
  background-color: ${theme.primaryDark};
  position: fixed;
  width: 100%;
  padding: 0.5rem 1rem;

  > * {
    margin: 0 0.5rem;

    &:first-child {
      margin: 0 1rem 0 0;
    }

    &:last-child {
      margin: 0 0 0 0.5rem;
    }
  }
`;

RecordingControls.propTypes = {
  className: PropTypes.string.isRequired,
  recordingEnabled: PropTypes.bool.isRequired,
  disableForward: PropTypes.bool,
  disableClear: PropTypes.bool,
  onToggleRecording: PropTypes.func.isRequired,
  onClearRecording: PropTypes.func.isRequired,
  onBack: PropTypes.func.isRequired,
  onForward: PropTypes.func.isRequired
};

const mapStateToProps = (state) => {
  const pastSize = state.todos.get('past').size;
  const futureSize = state.todos.get('future').size;

  return {
    recordingEnabled: state.todos.get('recordingEnabled'),
    disableForward: futureSize === 0,
    disableClear: pastSize === 0 && futureSize === 0
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onToggleRecording: () => dispatch(toggleRecording()),
    onClearRecording: () => dispatch(clearRecording()),
    onBack: () => dispatch(back()),
    onForward: () => dispatch(forward())
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(StyledRecordingControls);
