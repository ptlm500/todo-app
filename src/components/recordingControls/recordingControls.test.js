import React from 'react';
import { renderWithIntl } from '../../../test/utils';
import { StyledRecordingControls } from './RecordingControls';
import 'jest-styled-components';

const defaultProps = {
  recordingEnabled: true,
  disableForward: true,
  disableClear: true,
  onToggleRecording: jest.fn(),
  onClearRecording: jest.fn(),
  onBack: jest.fn(),
  onForward: jest.fn()
};

beforeEach(() => {
  jest.clearAllMocks();
});

describe('RecordingControls', () => {
  it('renders StyledRecordingControls matching the snapshot when recording is enabled', () => {
    const tree = renderWithIntl(<StyledRecordingControls {...defaultProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders StyledRecordingControls matching the snapshot when recording is disabled', () => {
    const props = {
      ...defaultProps,
      recordingEnabled: false
    };
    const tree = renderWithIntl(<StyledRecordingControls {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('passes the correct props to the toggle button', () => {
    const tree = renderWithIntl(<StyledRecordingControls {...defaultProps} />);

    const toggleButton = tree.root.findByProps({id: 'toggle-recording'});

    toggleButton.props.onClick();
    expect(defaultProps.onToggleRecording).toHaveBeenCalled();
  });

  it('passes the correct props to the clear button', () => {
    const props = {
      ...defaultProps,
      disableClear: false
    };
    const tree = renderWithIntl(<StyledRecordingControls {...props} />);

    const clearButton = tree.root.findByProps({id: 'clear'});

    expect(clearButton.props.disabled).toBeFalsy();

    clearButton.props.onClick();
    expect(defaultProps.onClearRecording).toHaveBeenCalled();
  });

  it('passes the correct props to the undo button', () => {
    const tree = renderWithIntl(<StyledRecordingControls {...defaultProps} />);

    const undoButton = tree.root.findByProps({id: 'undo'});

    undoButton.props.onClick();
    expect(defaultProps.onBack).toHaveBeenCalled();
  });

  it('passes the correct props to the redo button', () => {
    const props = {
      ...defaultProps,
      disableForward: false
    };
    const tree = renderWithIntl(<StyledRecordingControls {...props} />);

    const redoButton = tree.root.findByProps({id: 'redo'});

    expect(redoButton.props.disabled).toBeFalsy();

    redoButton.props.onClick();
    expect(defaultProps.onForward).toHaveBeenCalled();
  });
});
