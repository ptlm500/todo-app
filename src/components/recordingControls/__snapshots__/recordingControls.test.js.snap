// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordingControls renders StyledRecordingControls matching the snapshot when recording is disabled 1`] = `
.c3 {
  background: #3F51B5;
  border-radius: 0.5rem;
  border: 2px solid #3F51B5;
  cursor: pointer;
  color: #FFFFFF;
  height: 2rem;
  padding: 0.25rem 1rem;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  fill: #FFFFFF;
  background: #FFFFFF;
  border: 2px solid #3F51B5;
  color: #3F51B5;
  fill: #3F51B5;
}

.c3 > svg {
  width: 1.5rem;
  width: 1.5rem;
  fill: currentColor;
}

.c3:hover:not([disabled]) {
  background: #303F9F;
  border: 2px solid #303F9F;
}

.c3:disabled {
  border: 2px solid #BDBDBD;
  background: #BDBDBD;
  color: #BDBDBD;
  fill: #3F51B5;
}

.c3:hover:not([disabled]) {
  background: #3F51B5;
  color: #FFFFFF;
  fill: #FFFFFF;
}

.c3:disabled {
  background: #FFFFFF;
  border: 2px solid #BDBDBD;
  cursor: auto;
}

.c4 {
  background: #F44336;
  border-radius: 0.5rem;
  border: 2px solid #F44336;
  cursor: pointer;
  color: #FFFFFF;
  height: 2rem;
  padding: 0.25rem 1rem;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  fill: #FFFFFF;
  background: #FFFFFF;
  border: 2px solid #D32F2F;
  color: #D32F2F;
  fill: #D32F2F;
}

.c4 > svg {
  width: 1.5rem;
  width: 1.5rem;
  fill: currentColor;
}

.c4:hover:not([disabled]) {
  background: #F44336;
  border: 2px solid #F44336;
}

.c4:disabled {
  border: 2px solid #BDBDBD;
  background: #BDBDBD;
  color: #BDBDBD;
  fill: #3F51B5;
}

.c4:hover:not([disabled]) {
  background: #D32F2F;
  color: #FFFFFF;
  fill: #FFFFFF;
}

.c4:disabled {
  background: #FFFFFF;
  border: 2px solid #BDBDBD;
  cursor: auto;
}

.c1 {
  display: inline-block;
  width: 12px;
  height: 12px;
}

.c2 {
  -webkit-animation: none;
  animation: none;
  fill: #BDBDBD;
  width: 100;
  height: 100;
  -webkit-transition: 0.1s;
  transition: 0.1s;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  left: 0;
  bottom: 0;
  background-color: #303F9F;
  position: fixed;
  width: 100%;
  padding: 0.5rem 5%;
}

.c0 > * {
  margin: 0 0.5rem;
}

.c0 > *:first-child {
  margin: 0 1rem 0 0;
}

.c0 > *:last-child {
  margin: 0 0 0 0.5rem;
}

<div
  className="c0"
>
  <svg
    className="c1"
    viewBox="0 0 100 100"
  >
    <circle
      className="c2"
      cx="50"
      cy="50"
      disabled={true}
      fill="#D32F2F"
      r="50"
    />
  </svg>
  <button
    className="sc-AxjAm c3"
    id="toggle-recording"
    onClick={[MockFunction]}
  >
    Start
  </button>
  <button
    className="sc-AxjAm c4"
    disabled={true}
    id="clear"
    onClick={[MockFunction]}
  >
    Clear
  </button>
  <button
    className="sc-AxjAm c3"
    id="undo"
    onClick={[MockFunction]}
  >
    <svg
      height="9px"
      version="1.1"
      viewBox="0 0 21 9"
      width="21px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Undo
      </title>
      <g
        fill="none"
        fillRule="evenodd"
        stroke="none"
        strokeWidth="1"
      >
        <g
          transform="translate(-238.000000, -1533.000000)"
        >
          <g
            transform="translate(100.000000, 1428.000000)"
          >
            <g
              id="Outlined-/-Content-/-undo"
              transform="translate(136.000000, 98.000000)"
            >
              <g>
                <polygon
                  id="Path"
                  points="0 0 24 0 24 24 0 24"
                />
                <path
                  d="M12.5,8 C9.85,8 7.45,8.99 5.6,10.6 L2,7 L2,16 L11,16 L7.38,12.38 C8.77,11.22 10.54,10.5 12.5,10.5 C16.04,10.5 19.05,12.81 20.1,16 L22.47,15.22 C21.08,11.03 17.15,8 12.5,8 Z"
                  fill="currentColor"
                />
              </g>
            </g>
          </g>
        </g>
      </g>
    </svg>
  </button>
  <button
    className="sc-AxjAm c3"
    disabled={true}
    id="redo"
    onClick={[MockFunction]}
  >
    <svg
      height="9px"
      version="1.1"
      viewBox="0 0 21 9"
      width="21px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Redo
      </title>
      <g
        fill="none"
        fillRule="evenodd"
        stroke="none"
        strokeWidth="1"
      >
        <g
          transform="translate(-747.000000, -1533.000000)"
        >
          <g
            transform="translate(100.000000, 1428.000000)"
          >
            <g
              id="Outlined-/-Content-/-redo"
              transform="translate(646.000000, 98.000000)"
            >
              <g>
                <polygon
                  id="Path"
                  points="0 0 24 0 24 24 0 24"
                />
                <path
                  d="M18.4,10.6 C16.55,8.99 14.15,8 11.5,8 C6.85,8 2.92,11.03 1.54,15.22 L3.9,16 C4.95,12.81 7.95,10.5 11.5,10.5 C13.45,10.5 15.23,11.22 16.62,12.38 L13,16 L22,16 L22,7 L18.4,10.6 Z"
                  fill="currentColor"
                />
              </g>
            </g>
          </g>
        </g>
      </g>
    </svg>
  </button>
</div>
`;

exports[`RecordingControls renders StyledRecordingControls matching the snapshot when recording is enabled 1`] = `
.c3 {
  background: #3F51B5;
  border-radius: 0.5rem;
  border: 2px solid #3F51B5;
  cursor: pointer;
  color: #FFFFFF;
  height: 2rem;
  padding: 0.25rem 1rem;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  fill: #FFFFFF;
  background: #FFFFFF;
  border: 2px solid #3F51B5;
  color: #3F51B5;
  fill: #3F51B5;
}

.c3 > svg {
  width: 1.5rem;
  width: 1.5rem;
  fill: currentColor;
}

.c3:hover:not([disabled]) {
  background: #303F9F;
  border: 2px solid #303F9F;
}

.c3:disabled {
  border: 2px solid #BDBDBD;
  background: #BDBDBD;
  color: #BDBDBD;
  fill: #3F51B5;
}

.c3:hover:not([disabled]) {
  background: #3F51B5;
  color: #FFFFFF;
  fill: #FFFFFF;
}

.c3:disabled {
  background: #FFFFFF;
  border: 2px solid #BDBDBD;
  cursor: auto;
}

.c4 {
  background: #F44336;
  border-radius: 0.5rem;
  border: 2px solid #F44336;
  cursor: pointer;
  color: #FFFFFF;
  height: 2rem;
  padding: 0.25rem 1rem;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  fill: #FFFFFF;
  background: #FFFFFF;
  border: 2px solid #D32F2F;
  color: #D32F2F;
  fill: #D32F2F;
}

.c4 > svg {
  width: 1.5rem;
  width: 1.5rem;
  fill: currentColor;
}

.c4:hover:not([disabled]) {
  background: #F44336;
  border: 2px solid #F44336;
}

.c4:disabled {
  border: 2px solid #BDBDBD;
  background: #BDBDBD;
  color: #BDBDBD;
  fill: #3F51B5;
}

.c4:hover:not([disabled]) {
  background: #D32F2F;
  color: #FFFFFF;
  fill: #FFFFFF;
}

.c4:disabled {
  background: #FFFFFF;
  border: 2px solid #BDBDBD;
  cursor: auto;
}

.c1 {
  display: inline-block;
  width: 12px;
  height: 12px;
}

.c2 {
  -webkit-animation: ehXoqr infinite 4s ease-in-out;
  animation: ehXoqr infinite 4s ease-in-out;
  fill: #D32F2F;
  width: 100;
  height: 100;
  -webkit-transition: 0.1s;
  transition: 0.1s;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  left: 0;
  bottom: 0;
  background-color: #303F9F;
  position: fixed;
  width: 100%;
  padding: 0.5rem 5%;
}

.c0 > * {
  margin: 0 0.5rem;
}

.c0 > *:first-child {
  margin: 0 1rem 0 0;
}

.c0 > *:last-child {
  margin: 0 0 0 0.5rem;
}

<div
  className="c0"
>
  <svg
    className="c1"
    viewBox="0 0 100 100"
  >
    <circle
      className="c2"
      cx="50"
      cy="50"
      disabled={false}
      fill="#D32F2F"
      r="50"
    />
  </svg>
  <button
    className="sc-AxjAm c3"
    id="toggle-recording"
    onClick={[MockFunction]}
  >
    Stop
  </button>
  <button
    className="sc-AxjAm c4"
    disabled={true}
    id="clear"
    onClick={[MockFunction]}
  >
    Clear
  </button>
  <button
    className="sc-AxjAm c3"
    id="undo"
    onClick={[MockFunction]}
  >
    <svg
      height="9px"
      version="1.1"
      viewBox="0 0 21 9"
      width="21px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Undo
      </title>
      <g
        fill="none"
        fillRule="evenodd"
        stroke="none"
        strokeWidth="1"
      >
        <g
          transform="translate(-238.000000, -1533.000000)"
        >
          <g
            transform="translate(100.000000, 1428.000000)"
          >
            <g
              id="Outlined-/-Content-/-undo"
              transform="translate(136.000000, 98.000000)"
            >
              <g>
                <polygon
                  id="Path"
                  points="0 0 24 0 24 24 0 24"
                />
                <path
                  d="M12.5,8 C9.85,8 7.45,8.99 5.6,10.6 L2,7 L2,16 L11,16 L7.38,12.38 C8.77,11.22 10.54,10.5 12.5,10.5 C16.04,10.5 19.05,12.81 20.1,16 L22.47,15.22 C21.08,11.03 17.15,8 12.5,8 Z"
                  fill="currentColor"
                />
              </g>
            </g>
          </g>
        </g>
      </g>
    </svg>
  </button>
  <button
    className="sc-AxjAm c3"
    disabled={true}
    id="redo"
    onClick={[MockFunction]}
  >
    <svg
      height="9px"
      version="1.1"
      viewBox="0 0 21 9"
      width="21px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Redo
      </title>
      <g
        fill="none"
        fillRule="evenodd"
        stroke="none"
        strokeWidth="1"
      >
        <g
          transform="translate(-747.000000, -1533.000000)"
        >
          <g
            transform="translate(100.000000, 1428.000000)"
          >
            <g
              id="Outlined-/-Content-/-redo"
              transform="translate(646.000000, 98.000000)"
            >
              <g>
                <polygon
                  id="Path"
                  points="0 0 24 0 24 24 0 24"
                />
                <path
                  d="M18.4,10.6 C16.55,8.99 14.15,8 11.5,8 C6.85,8 2.92,11.03 1.54,15.22 L3.9,16 C4.95,12.81 7.95,10.5 11.5,10.5 C13.45,10.5 15.23,11.22 16.62,12.38 L13,16 L22,16 L22,7 L18.4,10.6 Z"
                  fill="currentColor"
                />
              </g>
            </g>
          </g>
        </g>
      </g>
    </svg>
  </button>
</div>
`;
